File: Daily Challanges/2026-04-April/2026-04-18.py
def find_sum(arr, target):
    n = len(arr)

    def dfs(i, current_sum, path):
        if current_sum == target and len(path) >= 2:
            return path
        if i == n:
            return None
        res = dfs(i + 1, current_sum + arr[i], path + [arr[i]])
        if res:
            return res
        return dfs(i + 1, current_sum, path)

    result = dfs(0, 0, [])
    return result if result else "Sum not found"

print(find_sum([1, 3, 5, 7], 6))

File: Daily Challanges/2026-04-April/2026-04-18.js
function findSum(arr, target) {
  const n = arr.length;

  function dfs(i, currentSum, path) {
    if (currentSum === target && path.length >= 2) {
      return path;
    }
    if (i === n) {
      return null;
    }
    let res = dfs(i + 1, currentSum + arr[i], [...path, arr[i]]);
    if (res) {
      return res;
    }
    return dfs(i + 1, currentSum, path);
  }

  const result = dfs(0, 0, []);
  return result ? result : "Sum not found";
}

console.log(findSum([1, 3, 5, 7], 6));
